core/state_transition.go

```diff
@@ -300,10 +306,10 @@ func (st *StateTransition) preCheck() error {
 	}
 	if !msg.SkipFromEOACheck {
 		// Make sure the sender is an EOA
-		codeHash := st.state.GetCodeHash(msg.From)
-		if codeHash != (common.Hash{}) && codeHash != types.EmptyCodeHash {
+		code := st.state.GetCode(msg.From)
+		if 0 < len(code) && !bytes.HasPrefix(code, types.DelegationPrefix) {
```

Comment:
Ah this is where I got that check from. @holiman commented on my EOF branch that we shouldn't do `0 < len(code)` but rather `len(code) > 0`. Now the delegation prefix of 0xfe0100 also makes sense

Reply:
why is it better swapped?

Reply:
So, first of all, neither is wrong, per se. 

Using `variable < constant` is how we typically do it in go-ethereum, and I think it's more standard across the board. 

I think the reason it's more common is that it's closer to most languages. 

> It's below five degrees outside!
 
I.e: `it < 5 `.

 Whereas you wouldn't not say: 

> Five degrees is warmer than it is outside!

Your code is now saying 

> If zero is smaller than the length of the code ... 

As opposed to

> If the length of the code is larger than zero .. 

---------------

There's an argument that can be made for doing it the opposite way, particularly when it comes to equality-check. 
`if 0 == x` vs `if x == 0`. If the programmer makes a typo, then `if 0=x` will be a compilation failure, whereas `if x=0` will not. I don't think it's particularly relevant in our code.